File: software/examples/python/example_hello_world.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_lcd_20x4 import LCD20x4

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    lcd = LCD20x4(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Turn backlight on
    lcd.backlight_on()

    # Write "Hello World"
    lcd.write_line(0, 0, 'Hello World')

    raw_input('Press key to exit\n') # Use input() in Python 3

File: software/examples/javascript/ExampleHelloWorld.js
var IPConnection = require('Tinkerforge/IPConnection');
var BrickletLCD20x4 = require('Tinkerforge/BrickletLCD20x4');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'SCD32';// Change to your UID

var ipcon = new IPConnection();// Create IP connection
var lcd = new BrickletLCD20x4(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
    // Turn backlight on
    lcd.backlightOn();
    // Write "Hello World"
    lcd.writeLine(0, 0, 'Hello World');     
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

